### ISSUE TITLE
'html' subcommand doesn't run on mac os x

### ISSUE THREAD

**Issue by bentxt** [PerBothner/DomTerm#66]
this command works

`./bin/domterm help`

html is displayed properly in the pager

`./bin/domterm html share/domterm/help/domterm.html`

this doesn't work, the cursor is stuck on new line and and cannot be freed or aborter. Domterm hangs and has to be closed.

This was tested on firefox and chrome on mac os x

```
sw_vers
ProductName:    Mac OS X
ProductVersion: 10.15

```

**Comment by PerBothner** [comment]
On 11/12/19 2:53 AM, ben.txt wrote:
> this command works
> 
> |./bin/domterm help|
> 
> html is displayed properly in the pager
> 
> |./bin/domterm html share/domterm/help/domterm.html|
> 
> this doesn't work, the cursor is stuck on new line and and cannot be freed or aborter. Domterm hangs and has to be closed.

That is a useful clue: 'domterm help' is executed in the "client" domterm (the
"domterm help" process), while "domterm html" sends the request (using a Unix domain
socket) over to the domterm server process.  (The reason is somewhat complicated, but has
mostly to do with resolving relative image references in a secure way.)

Perhaps a difference on Mac in how file descriptors are sent over a socket?

-- 
	--Per Bothner
per@bothner.com   http://per.bothner.com/

**Comment by PerBothner** [comment]
I got myself a used MacBook Air, so I've been able to build DomTerm and test it.

This problem is related to Unix/domain sockets, used to communicate between the `html` command (the "client") and the background "server".  The problem is when the client calls `connect`.  The server (which is waiting in `accept`) does make the connection (the `accept` returns) but in the client the `connect` call never returns. The server continues, but without the data it expects from the client.

This logic works fine on Linux (including WSL), but the networking on MacOS/Darwin/BSD is different, and I haven't figured out how.  Possibly something relating to blocking vs non-blocking I/O?
